sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -655,16 +688,53 @@ private boolean isMoreRecent(PackageInfo packageInfo, ReleaseDetails releaseDeta
     }
 
     /**
+     * Check if dialog should be restored in the new activity. Hiding previous dialog version if any.
+     *
+     * @param alertDialog existing dialog if any, always returning true when null.
+     * @return true if a new dialog should be displayed, false otherwise.
+     */
+    private boolean shouldRefreshDialog(@Nullable AlertDialog alertDialog) {
+
+        /* We could be in another activity now, refresh dialog. */
+        if (alertDialog != null) {
+
+            /* Nothing to if resuming same activity with dialog already displayed. */
+            if (alertDialog.isShowing()) {
+                if (mForegroundActivity == mLastActivityWithDialog.get()) {
+                    MobileCenterLog.debug(LOG_TAG, "Previous dialog still shown in same activity.");
```

Comment:
Previous dialog **is still shown**/**is still being shown**  in the same activity.